torch/_higher_order_ops/flex_attention.py

```diff
@@ -1146,17 +1150,44 @@ def flex_attention_backward_fake_tensor_mode(
     torch.Tensor, torch.Tensor, torch.Tensor, tuple[Optional[torch.Tensor], ...]
 ]:
     with mode:
-        grad_query = torch.empty_like(query)
-        grad_key = torch.empty_like(key)
-        grad_value = torch.empty_like(value)
+        Bq = query.size(0)
+        Bkv = key.size(0)
+        Hkv = key.size(1)
+        seq_len_kv = key.size(2)
+        qk_head_dim = key.size(3)
+
+        grad_query = torch.empty_like(query, memory_format=torch.preserve_format)
+        # zeros_and_scatter creates a contiguous zeros tensor -> contiguous_format
         grad_score_mod_captured = tuple(
             [
-                torch.empty_like(buffer)
+                torch.empty_like(buffer, memory_format=torch.contiguous_format)
                 if isinstance(buffer, torch.Tensor) and buffer.requires_grad
                 else None
                 for buffer in score_mod_other_buffers
             ]
         )
+
+        broadcasted_grad_key = torch.empty_strided(
+            (Bq, Hkv, seq_len_kv, qk_head_dim),
+            key.stride(),
+            device=key.device,
+            dtype=key.dtype,
+        )
+
+        broadcasted_grad_value = torch.empty_strided(
+            (Bq, *value.size()[1:]),
+            value.stride(),
+            device=value.device,
+            dtype=value.dtype,
+        )
+
```

Comment:
For my own understanding, is the below the multi query case

Reply:
This isn't multiquery its just for broadcasting along Bastchdim which is useful for paged attention